File: xide-ui/src/main/java/org/xast/xide/ui/state/tool/SettingsTool.java
package org.xast.xide.ui.state.tool;

import org.xast.xide.ui.MainFrame;
import org.xast.xide.ui.dialog.SettingsDialog;

import lombok.AllArgsConstructor;

@AllArgsConstructor
public class SettingsTool implements Tool {
    private final MainFrame frame;

    @Override
    public void show() {
        new SettingsDialog(frame).setVisible(true);
    }
}

File: xide-standard-plugins/xide-settings-plugin/src/main/kotlin/org/xast/xide/settings_plugin/SettingsTool.kt
package org.xast.xide.settings_plugin

import javax.swing.JFrame
import org.xast.xide.core.plugin.tool.Tool

class SettingsTool(val frame: JFrame) : Tool {
    override fun show() {
        SettingsDialog(this.frame).isVisible = true
    }
}
